axum/src/extract/ws.rs

```diff
@@ -530,6 +530,13 @@ impl WebSocket {
     }
 }
 
+impl FusedStream for WebSocket {
+    /// Returns true if the websocket has been terminated.
```

Comment:
This is just a nit for your future reference, but doc comments on trait implementations are pretty hard to find to the point that I'd argue they're mostly useless. They don't hurt though.